packages/cli/templates/python/graph/src/main.py

```diff
@@ -0,0 +1,214 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+import asyncio
+import logging
+import os
+
+from azure.core.exceptions import ClientAuthenticationError
+from microsoft.teams.api import MessageActivity
+from microsoft.teams.apps import ActivityContext, App, AppOptions, ErrorEvent, SignInEvent
+from microsoft.teams.graph import get_graph_client
+from msgraph.generated.users.item.messages.messages_request_builder import (  # type: ignore
+    MessagesRequestBuilder,
+)
+
+logger = logging.getLogger(__name__)
+
+# Create app with OAuth connection
+app_options = AppOptions(default_connection_name=os.getenv("CONNECTION_NAME", "graph"))
+app = App(app_options)
+
+
+async def get_authenticated_graph_client(ctx: ActivityContext[MessageActivity]):
+    """
+    Helper function to handle authentication and create Graph client using Token pattern.
+
+    Returns:
+        Graph client if successful, None if authentication failed.
+    """
+    # Check if user is signed in
+    if not ctx.is_signed_in:
+        await ctx.send("🔐 Please sign in first to access Microsoft Graph.")
+        await ctx.sign_in()
+        return None
+
+    try:
+        # Create Graph client using the user token
+        return get_graph_client(ctx.user_token)
+
+    except Exception as e:
+        ctx.logger.error(f"Failed to create Graph client: {e}")
+        await ctx.send("🔐 Failed to create authenticated client. Please try signing in again.")
+        await ctx.sign_in()
+        return None
+
+
+@app.on_message_pattern("signin")
```

Comment:
i'd say let's pear this sample down to simple things:
1. Profile
2. message_pattern('signout')

Complex templates have lead to more confusion, and questions. Let's keep it simple :)